docs/documentation/advanced/unkown-fields.fsx

```diff
@@ -0,0 +1,302 @@
+(**
+---
+layout: standard
+title: Unkown fields
+---
+*)
+
+(*** hide ***)
+
+#I "../../../src/bin/Debug/netstandard2.0"
+#r "Thoth.Json.dll"
+
+open Thoth.Json
+
+(**
+
+It can happens that you receive a JSON but you don't know which fields are going to be present.
+
+Consider the following JSON:
+
+```json
+{
+    "ts": "2020-01-01T00:00:00Z",
+    "EUR_PLN": { "rate": "4.55" },
+    "GBP_PLN": { "error": "Rate is not available at the moment" },
+    "USD_PLN": { "rate": "4.01" }
+}
+```
+
+In this example, we know that:
+
+- The `ts` field required
+- There is an unkown number of `Rate` fields which consists of:
+    - The key property which has the format:
+        1. `sourceCurrency`
+        2. `_`
+        3. `targetCurrency`
+    - The value should have a `rate` field which is a `decimal`
+
+We are now going to write a docuder capable of handling such a JSON.
+
+## Helpers
+
+First, we are going to define some helpers to make the code more readable.
+*)
+
+module Option =
+
+    /// Return the value if there is one, otherwise throw an exception
+    let unwrap (value : 'T option) =
+        match value with
+        | Some value -> value
+        | None -> failwith "Option.unwrap: None"
+
+module List =
+
+    /// Returns all the elements in the list which have a value
+    let unwrapOption (list : ('T option) list) =
+        list
+        |> List.filter Option.isSome
+        |> List.map Option.unwrap
```

Comment:
Why would one do this instead of `List.choose id`… it's basically a flat mapping through the list, but instead of single list elements or empty elements we have optionals.

Reply:
```fsharp
> let xs = [ Some 1; Some 2; None; Some 3];;
val xs: int option list = [Some 1; Some 2; None; Some 3]

> xs |> List.choose id;;
val it: int list = [1; 2; 3]
```

Reply:
I didn't know about `List.choose id`, will change it for that